mod-source-record-storage-server/src/main/java/org/folio/verticle/MarcIndexersVersionDeletionVerticle.java

```diff
@@ -25,36 +34,72 @@ public class MarcIndexersVersionDeletionVerticle extends AbstractVerticle {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
-  private RecordDao recordDao;
-  private TenantDataProvider tenantDataProvider;
+  private static final ZoneId ZONE_ID = ZoneId.systemDefault();
+
+  private final RecordDao recordDao;
+  private final TenantDataProvider tenantDataProvider;
+  private List<LocalTime> scheduleTimes;
+
+  @Value("${srs.marcIndexers.delete.time:01:00}")
+  private String deleteTimes;
+
+  @Value("${srs.marcIndexers.delete.oneTimeLimit:100000}")
+  private Integer oneTimeLimit;
+
+  private Future<Boolean> currentDeletion = Future.succeededFuture();
 
   @Autowired
   public MarcIndexersVersionDeletionVerticle(RecordDao recordDao, TenantDataProvider tenantDataProvider) {
     this.recordDao = recordDao;
     this.tenantDataProvider = tenantDataProvider;
   }
 
-  @Value("${srs.marcIndexers.delete.interval.seconds:1800}")
-  private int interval;
-
-  private Future<Boolean> currentDeletion = Future.succeededFuture();
-
   @Override
   public void start(Promise<Void> startFuture) {
-    vertx.setPeriodic(interval * 1000L, id -> {
-      if (currentDeletion.isComplete()) {
-        currentDeletion = deleteOldMarcIndexerVersions();
-      } else {
-        LOGGER.info("Previous marc_indexers old version deletion still ongoing");
-      }
-    });
+    this.scheduleTimes = Arrays.stream(deleteTimes.split(","))
+      .map(LocalTime::parse)
+      .sorted()
+      .collect(Collectors.toList());
+    scheduleNextTask(vertx, this::deleteTask);
     startFuture.complete();
   }
 
+  private void scheduleNextTask(Vertx vertx, Runnable task) {
```

Comment:
I am thinking a bunch of this code could be replaced by doing Vertx.periodicTimer at the start time. Something to consider.

Reply:
I think it would be better to move this processing to night time and we can set the time using commas like: srs.marcIndexers.delete.time: 01:00,02:00

Reply:
I am not disputing moving to specific time. I am saying after starting a specific time, using Vertx.periodic with 1 day duration would be less code. Looking at the code, I can't tell if it works since a lot of the api that is not commonly used by me. This is not a mandatory change, just a suggestion.